src/containers/JoinUs.tsx

```diff
@@ -3,32 +3,44 @@
 import React, { useState } from 'react';
 import Image from 'next/image';
 
-import CInput from '../components/Cinput';
-import CButton from '../components/Cbutton';
+import toast from '../components/CToast';
+import CInput from '../components/CInput';
+import CButton from '../components/CButton';
 import { ExternalPages } from '../constants/externalPages';
 
 import fetch from '../utils/request';
 
-import JoinDancing from 'public/images/joinDancing.png';
+import joinDancing from 'public/images/joinDancing.png';
 
-const JoinUs = () => {
-  const [value, setValue] = useState('');
-
-  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value);
-  };
+const validateEmail = (email: string): boolean => {
+  const emailRegex = /.+\@.+\..+/;
+  return emailRegex.test(email);
+};
 
+const JoinUs = () => {
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
+    const email = event.target.input.value;
+    if (!validateEmail(email)) {
+      toast('error', 'Enter a valid email address');
+      return;
+    }
+
     fetch(ExternalPages.FLUXITY_API + '/subscribe', {
       method: 'POST',
       body: JSON.stringify({
-        email: value,
+        email: email,
       }),
       headers: {
         'Content-Type': 'application/json',
       },
-    });
+    })
+      .then(() => {
+        toast('success', 'Subscription successful!');
+      })
+      .catch((error) => {
+        toast('error', error.data.message + '.');
```

Comment:
if (error?.data?.message , then toast it, if it's undefined, toast something default